Task05/src/citiesMap.cpp

```diff
@@ -0,0 +1,134 @@
+#include "citiesMap.h"
+#include <iostream>
+#include <string>
+#include <map>
+#include <algorithm>
+#include <fstream>
+#include <sstream>
+#include <vector>
+ 
+
+void citiesMap::insert(std::string cityName, double x, double y)
+{	 
+	coordinate newCity = { x, y };
+
+	this->cityMap.insert({ cityName,newCity });
+}
+
+ 
+double calculateEuclideanDistance(double x0, double y0, double x1, double y1) {
```

Comment:
there is std::function in the STL library that you can use and not a pointer to function